skimage/color/colorconv.py

```diff
@@ -1077,6 +1084,8 @@ def lab2xyz(lab, illuminant="D65", observer="2", *, channel_axis=-1):
         warn(f'Color data out of range: Z < 0 in {invalid[0].size} pixels',
```

Comment:
I guess what I find really weird about the logic as it current stands is that:

1. For all valid images, this costs: a single comparison to 0 and an array traversal.
2. For invlida images, this costs: a single comparison to 0 and an array traversal, followed by a second comaprison to 0 and a second array array traversal.


IF the code is refactored to:

```
invalid = np.nonzero(z < 0)
if len(invalid) != 0:
    warn(...)
```
then it would be both shorter, and cost only:

- a comparison to 0, and a single array traversal in both cases.